### ISSUE TITLE
Discussion: tidy up post 2.1.0

### ISSUE THREAD

**Issue by znicholls** [openclimatedata/pymagicc#166]
So I think for 2.1.0 we should aim to:

- [ ] integrate with openscm 
- [ ] use of some variant of pyam dataframe

Then I think we should focus on refactoring things, in particular:

- [ ] tidying up docs
- [ ] re-write any untidy code
- [ ] split the tests into unit and integration so it becomes easier to see what's going on and develop in future

**Comment by znicholls** [comment]
yes but if we break that API (i.e. rename `pymagicc.api` to `pymagicc.core`) then we should really bump to 3.0 if we're going to follow semantic versioning properly. Plan b is we change it before the release?

**Comment by rgieseke** [comment]
Both possible, just wanted to note down the idea. We can also provide an alias and deprecation warning but that's probably overkill given we have hardly any users (and need to fix the MAGICC7 pipeline ourselves).
